core/iwasm/common/wasm_application.c

```diff
@@ -685,6 +685,10 @@ execute_func(WASMModuleInstanceCommon *module_inst, const char *name,
         goto fail;
     }
 
+    if (argv && type->result_count == 1 && type->types[type->param_count] == VALUE_TYPE_I32)
+        /* copy the return value */
+        *(int *)argv = (int)argv1[0];
```

Comment:
This only works for i32 type, if to support writing return values back to the array, we had better support all the types and support multiple return values, but it will lead to a lot of changes (e.g. main.c under product-mini) and API incompatibility, e.g., in 32-bit platform, there is only one argument and the size of argv is 4 bytes, but the return type is i64 and is 8 bytes, then developer had to pass larger argv array to this function. Could you use wasm_runtime_call_wasm instead?